src/x86/sse.asm

```diff
@@ -24,8 +24,8 @@ SECTION .text
     ; Multiply and shift using scalar code
     mov             scaled, [scaleq]
     imul               rax, scaleq
-    add                rax, 0x800
```

Comment:
Maybe I'm blind but I'm missing the corresponding rust change.

Reply:
Also I don't understand why we would want to lose precision at this stage if we're not currently overflowing anything.

Reply:
The rust code always calculates the round from `DIstortionScale::SHIFT` as `1 << Self::SHIFT >> 1`.
https://github.com/barrbrain/rav1e/blob/20eacc69/src/rdo.rs#L573-L579
https://github.com/barrbrain/rav1e/blob/20eacc69/src/rdo.rs#L626

Reply:
I don't think we are losing precision here, the change in shift matches the change in scale of the input. We have enough headroom for the additional precision.
However, I can see the argument for accumulating before shifting. Neither the rust nor assembly code do that yet.

Reply:
Ah, there is a TODO specifically for deferring rounding in the assembly:
https://github.com/xiph/rav1e/blob/8c260d2c/src/x86/sse.asm#L42-L46

Reply:
I drafted a compromise that avoided introducing overflow: https://github.com/barrbrain/rav1e/commit/0a50b08
AWCY results [on top of commits so far](https://beta.arewecompressedyet.com/?job=fixed-point%402022-09-05T05%3A19%3A26.619Z&job=fixed-point%402022-09-05T14%3A44%3A04.983Z):
|  PSNR Y | PSNR Cb | PSNR Cr | CIEDE2000 |   e  | MS-SSIM | PSNR-HVS Y | PSNR-HVS Cb | PSNR-HVS Cr | PSNR-HVS |    VMAF | VMAF-NEG |
|    ---: |    ---: |    ---: |      ---: |   ---: |    ---: |       ---: |        ---: |        ---: |     ---: |    ---: |     ---: |
| -0.1024 | -0.5120 | -0.4848 |   -0.4979 | 0.1456 |  0.1729 |    -0.0325 |     -0.6291 |     -0.5239 |  -0.1197 | -0.1653 |  -0.1696 |